doc/md/lint/analyzers.md

```diff
@@ -164,4 +182,41 @@ zero (default) value. For example:
 ALTER TABLE t ADD COLUMN c int NOT NULL;
 // highlight-next-line
 -- Append column `c` to all existing rows with the value 0.
+```
+
+#### LT101 {#LT101}
+
+Modifying a nullable column to non-nullable without setting a `DEFAULT` might fail in case it contains `NULL` values.
+The solution is one of the following:
+
+1\. Set a `DEFAULT` value on the modified column:
+
+```sql {2}
+-- create "new_users" table
+CREATE TABLE `new_users` (`a` int NOT NULL DEFAULT 1);
+-- copy rows from old table "users" to new temporary table "new_users"
+INSERT INTO `new_users` (`a`) SELECT IFNULL(`a`, 1) FROM `users`;
+-- drop "users" table after copying rows
+DROP TABLE `users`;
+-- rename temporary table "new_users" to "users"
+ALTER TABLE `new_users` RENAME TO `users`;
+```
+
+2\. Backfill `NULL` values with a default value:
+
+```sql {1-2}
+-- backfill previous rows
+UPDATE `users` SET `a` = 1 WHERE `a` IS NULL;
+-- disable the enforcement of foreign-keys constraints
+PRAGMA foreign_keys = off;
+-- create "new_users" table
+CREATE TABLE `new_users` (`a` int NOT NULL);
```

Comment:
the SQLite example is more complicated than other DBs here, maybe use another DB since it diverts focus

Reply:
This is an SQLite-specific example using the "temporary table" because SQLite is treated a bit differently. See https://www.sqlite.org/lang_altertable.html#making_other_kinds_of_table_schema_changes.

That's why we have MF104 and LT101 checks.